lib/time_util.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+class TimeUtil
+
+  # Normalize a timestamp to a more accurate time based on an offset of sent time
+  # vs now time.
+  #
+  # Return a unix time (ms since epoch)
+  #
+  # see https://images.zenhubusercontent.com/5c75601cce38d251a29ce098/bf41aa5b-c908-4b08-a738-5e940d007170
+  def self.device_adjust(normalize_at:, sent_at:)
+    offset_secs = Time.now - Time.parse(sent_at)
+    Time.parse(normalize_at) - offset_secs
```

Comment:
I think this should be a `+`.

Assume the following:
* `client_clock_occurred_at = 42`
* `client_clock_sent_at = 45`
* `env[:received_at] = 38`

The assumption in the book is that network travel time is negligible, meaning that `client_clock_sent_at` and `received_at` are the same actual point in time.  The book says to compute the offset as `offset = received_at - client_clock_sent_at`, which for us gives `38 - 45 = -7`.  Our server time is 7 seconds behind the client clock.  Since the client clock says the event occurred at time `42`, in server time that should be 7 seconds before which is `35`.  ... so we want:

`occurred_at = 35 = 42 + (-7) = client_clock_occurred_at + offset`

So `+` not `-`.

The test will need to be rewritten.

Reply:
Correct! :heavy_check_mark:

Reply:
done